datafusion/proto/src/lib.rs

```diff
@@ -480,14 +492,11 @@ mod roundtrip_tests {
                     ScalarValue::Float32(Some(2.0)),
                     ScalarValue::Float32(Some(1.0)),
                 ]),
-                DataType::List(new_box_field("level1", DataType::Float32, true)),
+                DataType::Float32,
```

Comment:
These are not valid `ScalarValue::List` structures -- specifically, the type of the field on a `List` is the type of the element (aka it is not DataType::List(type of element), which is what this test was previously doing).

You can see correct usages in the rest of the codebase https://github.com/apache/arrow-datafusion/search?q=new_list

for example

https://github.com/apache/arrow-datafusion/blob/e0a9fa347f7adb1e6d6157b537e44c574e88244a/datafusion/physical-expr/src/aggregate/array_agg.rs#L174-L183